api/src/nv_ingest_api/internal/extract/pdf/engines/pdfium.py

```diff
@@ -377,6 +378,94 @@ def _extract_page_elements(
     return extracted_page_elements
 
 
+# ------------------------------------------------------------------ #
+#  Render one page – now returns (page_idx, image, padding_offsets)  #
+# ------------------------------------------------------------------ #
+def _render_page_from_file(
+    pdf_source: Union[str, bytes],
+    page_idx: int,
+    size: Tuple[int, int],
+    execution_trace_log: Optional[List] = None,
+) -> Tuple[int, np.ndarray, Tuple[int, int]]:
+    """
+    Opens `pdf_source`, renders `page_idx`, and returns
+    (page_idx, image_array, padding_offsets).
+    """
+    doc = libpdfium.PdfDocument(pdf_source)
```

Comment:
Am I understanding correctly that if a PDF file was say 1,000 pages this would open/close that same file 1,000 times to get each individual page here?